### ISSUE TITLE
Ignore trailing if in scope for early exit

### ISSUE THREAD

**Issue by Stadly** [slevomat/coding-standard#574]
The [EarlyExit](https://github.com/slevomat/coding-standard#slevomatcodingstandardcontrolstructuresearlyexit-) sniff has the setting `ignoreStandaloneIfInScope`. What about ignoring `if`s that are the trailing part of the scope, and not only `if`s that are the only part of the scope? Then also such cases could be ignored:

```php
$errors = [];
foreach ($values as $value) {
    $error = $value->validate();

    if ($error !== null) {
        $errors[] = $error;
    }
}
```

In addition to the examples shown in https://github.com/slevomat/coding-standard/issues/371 and https://github.com/slevomat/coding-standard/issues/409.

Thank you for your consideration and great work!

**Comment by amorimjuliana** [comment]
Same issue here. For those cases, the early return makes the code less readable and more verbose:
```php
$errors = [];
foreach ($values as $value) {
    $error = $value->validate();

    if ($error === null) {
         continue;
    }

    $errors[] = $error;
}
```

**Comment by peldax** [comment]
Simple ignore would lead to mixture of EarlyExits and TrailingIfs accross codebase. In my opinion you should either require EarlyExit or require TrailingIf.

`requireTrailingIf`: true would always refactor the code to use TrailingIf, false to EarlyExit.

**Comment by jrfnl** [comment]
Suggestion: what about ignoring trailing `if`s if the content is only one line as early exit really doesn't make a difference in that case ?

```php
function foo() {
    $var = function_call();
    if ($var === true) {
        call_other_function();
    }
}

// Versus
function foo() {
    $var = function_call();
    if ($var !== true) {
        return;
    }

    call_other_function();
}
```

**Comment by kukulich** [comment]
New option implemented in https://github.com/slevomat/coding-standard/commit/055af01196f93a6e627ea03b7ec18a252c6efa3c

**Comment by Jurigag** [comment]
To be honest i like this new option, but don't like it allows no early exit if only one line - it should be more like one instruction, let's take a this example:

```php
 protected function addResultBasedOnCondition($budgetEnded, $timeEnded, $timeAlmostOver, $campaignUnderPerforming): void
    {
        if ($budgetEnded) {
            $timeAlertNotificationsCollection->addResult(
                new TimeAlertNotificationDto($campaignName, $brandName, Notification::TYPE_TIME_ALERT_BUDGET_ENDED)
            );
        }
        if ($timeEnded) {
            $timeAlertNotificationsCollection->addResult(
                new TimeAlertNotificationDto($campaignName, $brandName, Notification::TYPE_TIME_ALERT_TIME_ENDED)
            );
        }
        if ($timeAlmostOver) {
            $timeAlertNotificationsCollection->addResult(
                new TimeAlertNotificationDto($campaignName, $brandName, Notification::TYPE_TIME_ALERT_TIME_ALMOST_OVER)
            );
        }
        if ($campaignUnderPerforming) {
            $timeAlertNotificationsCollection->addResult(
                new TimeAlertNotificationDto($campaignName, $brandName, Notification::TYPE_TIME_ALERT_CAMPAIGN_IS_UNDERPERFORMING)
            );
        }
    }
```

It will require early exit on `if ($campaignUnderPerforming) {` even if `ignoreOneLineTrailingIf` is enabled, but if we format it like this:

```php
protected function addResultBasedOnCondition($budgetEnded, $timeEnded, $timeAlmostOver, $campaignUnderPerforming): void
    {
        if ($budgetEnded) {
            $timeAlertNotificationsCollection->addResult(
                new TimeAlertNotificationDto($campaignName, $brandName, Notification::TYPE_TIME_ALERT_BUDGET_ENDED)
            );
        }
        if ($timeEnded) {
            $timeAlertNotificationsCollection->addResult(
                new TimeAlertNotificationDto($campaignName, $brandName, Notification::TYPE_TIME_ALERT_TIME_ENDED)
            );
        }
        if ($timeAlmostOver) {
            $timeAlertNotificationsCollection->addResult(
                new TimeAlertNotificationDto($campaignName, $brandName, Notification::TYPE_TIME_ALERT_TIME_ALMOST_OVER)
            );
        }
        if ($campaignUnderPerforming) {
            $timeAlertNotificationsCollection->addResult(new TimeAlertNotificationDto($campaignName, $brandName, Notification::TYPE_TIME_ALERT_CAMPAIGN_IS_UNDERPERFORMING));
        }
    }
```

Then there is no error reported. Can this somehow be improved? @kukulich
